File: inbound/backend/src/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace backend.Models;

public class Product
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int ID { get; set; }

    [Required]
    public String? Name { get; set; }

    [Required]
    public int Quantity { get; set; } = 0;
}

File: inbound/frontend/lib/model/entities/product.dart
class Product {
  int id;
  String? name;
  int? quantity;
  Product(
      {required this.id,
        required this.name,
        required this.quantity});

  Product.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        quantity = json['quantity'];
}